code/game/objects/items/toys/toys.dm

```diff
@@ -567,3 +470,171 @@
+	icon_state = "bee"
+
+/obj/item/toy/plush/moth
+	name = "moth plush"
+	desc = "A plush doll of a bug."
+	icon_state = "moth"
+
+/obj/item/toy/plush/rock
+	name = "rock plush"
+	desc = "It says it is a plush on the tag, at least."
+	icon_state = "rock"
+
+/obj/item/toy/plush/therapy
+	name = "therapy plush"
+	desc = "A therapeutic toy to assist marines in recovering from mental and behavioral disorders after experiencing the trauma of battles."
+	icon_state = "therapy"
+
+/obj/item/toy/plush/therapy/red
+	name = "red therapy plush"
+	color = "#FC5274"
+
+/obj/item/toy/plush/therapy/blue
+	name = "blue therapy plush"
+	color = "#9EBAE0"
+
+/obj/item/toy/plush/therapy/green
+	name = "green therapy plush"
+	color = "#A3C940"
+
+/obj/item/toy/plush/therapy/orange
+	name = "orange therapy plush"
+	color = "#FD8535"
+
+/obj/item/toy/plush/therapy/purple
+	name = "purple therapy plush"
+	color = "#A26AC7"
+
+/obj/item/toy/plush/therapy/yellow
+	name = "yellow therapy plush"
+	color = "#FFE492"
+
+/obj/item/toy/plush/therapy/random_color
+	///Hexadecimal 0-F (0-15)
+	var/static/list/hexadecimal = list("0", "1", "2", "3" , "4", "5", "6", "7", "8", "9", "A", "B", "C", "D", "E", "F")
+
+/obj/item/toy/plush/therapy/random_color/New(loc, ...)
+	. = ..()
+	var/color_code = "#[pick(hexadecimal)][pick(hexadecimal)][pick(hexadecimal)][pick(hexadecimal)][pick(hexadecimal)][pick(hexadecimal)]" //This is dumb and I hope theres a better way I'm missing
+	color = color_code
+	desc = "A custom therapy plush, in a unique color. This one is labeled with \"#[color_code]\"."
+
+/obj/item/toy/plush/random_plushie //Not using an effect so it can fit into storage from loadout
+	name = "random plushie spawner"
+	desc = "You should not be seeing this"
+	/// Standard plushies for the spawner to pick from
+	var/list/plush_list = list(
+		/obj/item/toy/plush/farwa,
+		/obj/item/toy/plush/barricade,
+		/obj/item/toy/plush/bee,
+		/obj/item/toy/plush/shark,
+		/obj/item/toy/plush/moth,
+		/obj/item/toy/plush/rock,
+	)
+	///Therapy plushies left separately to not flood the entire list
+	var/list/therapy_plush_list = list(
+		/obj/item/toy/plush/therapy,
+		/obj/item/toy/plush/therapy/red,
+		/obj/item/toy/plush/therapy/blue,
+		/obj/item/toy/plush/therapy/green,
+		/obj/item/toy/plush/therapy/orange,
+		/obj/item/toy/plush/therapy/purple,
+		/obj/item/toy/plush/therapy/yellow,
+		/obj/item/toy/plush/therapy/random_color,
+	)
+
+/obj/item/toy/plush/random_plushie/Initialize(mapload, ...)
+	. = ..()
+	if(mapload) //Can't always create in initialization due to potential loadout issues if so
+		create_plushie()
+
+/obj/item/toy/plush/random_plushie/post_loadout_spawn(mob/living/carbon/human/user)
+	. = ..()
+	create_plushie()
+
+/obj/item/toy/plush/random_plushie/proc/create_plushie()
+	var/plush_list_variety = pick(60; plush_list, 40; therapy_plush_list)
+	var/random_plushie = pick(plush_list_variety)
+	new random_plushie(loc)
+	qdel(src)
```

Comment:
"potential loadout issues" ? can you elaborate ?

regarding the instanciation method - don't do that, instanciating it directly in storage bypasses all storage rules and handlers